planner/core/flat_plan.go

```diff
@@ -55,22 +55,34 @@ type FlatPhysicalPlan struct {
 type FlatPlanTree []*FlatOperator
 
 // GetSelectPlan skips Insert, Delete and Update at the beginning of the FlatPlanTree.
+// GetSelectPlan also skips foreign key check/cascade plan.
 // Note:
 //
-//	It returns a reference to the original FlatPlanTree, please avoid modifying the returned value.
+//	It returns a reference to the original FlatPlanTree and the offset, please avoid modifying the returned value.
 //	Since you get a part of the original slice, you need to adjust the FlatOperator.Depth and FlatOperator.ChildrenIdx when using them.
```

Comment:
```suggestion
//	It returns a reference to the original FlatPlanTree, please avoid modifying the returned value.
//	The second return value is the offset. Because the returned FlatPlanTree is a part of the original slice, you need to minus them by the offset when using the returned FlatOperator.Depth and FlatOperator.ChildrenIdx.
```